test/test_utilities.cpp

```diff
@@ -191,12 +191,15 @@ void Compare(Tensor& T1, Tensor& T2, float tol) {
   for (int k = 0; k < T_size; k++) {
     auto a = T1_data[k];
     auto b = T2_data[k];
-    if (a == 0) {
-      EXPECT_NEAR(a, b, tol);
-    } else {
-      auto rel = a - b;
-      auto rel_div = std::abs(rel / a);
-      EXPECT_TRUE(rel_div <= tol);
+    if (!(std::isnan(a) && std::isnan(b)) &&
+        !(std::isinf(a) && std::isinf(b))) {
```

Comment:
In case of nan or inf, we are not comparing them.

For positive and negative infinities, they might be comparable. hence we should check them. Not sure in the case of nan

For unit tests: you could create a tensor and populate it with Nan/inf maybe using this function: https://github.com/tensorflow/ngraph-bridge/blob/02e02434216b017a4879bfe4453312a937ef87a1/test/test_math_ops.cpp#L112